ci/src/config.rs

```diff
@@ -302,3 +306,46 @@ impl From<&OsStr> for GoogleApplicationCredentials {
         }
     }
 }
+
+/// A valid container image name without the image tag.
+///
+/// ```rust
+/// # use buildkite_hooks::config::ContainerImageName;
+/// use std::str::FromStr;
+///
+/// let img = ContainerImageName::from_str("ubuntu").unwrap();
+/// assert_eq!(&img as &str, "ubuntu");
+///
+/// let img_result = ContainerImageName::from_str("ubuntu:18.04");
+/// assert!(img_result.is_err());
+/// ```
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct ContainerImageName(String);
+
+impl std::fmt::Display for ContainerImageName {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result { self.0.fmt(f) }
+}
+
+impl std::str::FromStr for ContainerImageName {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        const PATTERN: &str = r"\A[a-zA-Z][a-zA-Z0-9./_\-]*\z";
+        let regex = regex::Regex::new(PATTERN).unwrap();
+        if regex.is_match(s) {
```

Comment:
```suggestion
        lazy_static! {
            static ref REGEX: Regex = regex::Regex::new(PATTERN).unwrap();
        }
        if REGEX.is_match(s) {
```

... as per the `regex` crate docs...